teps/0048-task-results-without-results.md

```diff
@@ -0,0 +1,425 @@
+variables, and their default values.
+
+In the following example, `pipelineTask` `add` has specified `results` with the default value. Now, `x` is referring
+to a task result of `add`. If `add` fails for some reason, finally task `multiply` continues execution with default
+`sum` which is `0`:
+
+```yaml
+kind: Pipeline
+spec:
+  tasks:
+    - name: add
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          default: 0
+  finally:
+    - name: multiply
+      taskRef:
+        name: multiply
+      params:
+        - name: x
+          value: $(task.add.results.sum)
+```
+
+### User Stories
+
+#### Story 1
+
+For a cleanup use case, `finally` task receives the project name, or the configuration name as a `task result` from
+the `acquire` task. If `acquire` task fails to acquire any resources, `cleanup` is skipped.
+
+#### Story 2
+
+Pipeline that builds a container image and deploys it. The pipeline accepts an image reference parameter.
+
+* The first task looks up in the registry if the image exists.
+
+* The second task builds the container images. It is guarded via `when` and only runs if the first task reports that the
+  image is missing.
+
+* The third task takes the image reference from a result of the second task and deploys the container.
+
+The pipeline authors sets a default value for the image reference of it's own image reference input param, so that
+if task2 is `skipped`, task3 can deploy the image that was passed to the pipeline by the `pipelineRun`.
+
+```yaml
+kind: Pipeline
+spec:
+  params:
+    - name: targetImage
+      description: the image to deploy (and build if missing)
+  tasks:
+    - name: lookup-image
+      taskRef:
+        name: search-image
+      params:
+        - name: imageRef
+          value: "$(params.targetImage)"
+    - name: build-image
+      when:
+        - input: "$(tasks.lookup-image.results.found)"
+          operator: in
+          values: ["False"]
+      taskRef:
+        name: build-image
+      params:
+        - name: imageRef
+          value: "$(params.targetImage)"
+      results:
+        - name: builtImage
+          default: "$(params.targetImage)"
+    - name: deploy-image
+      taskRef:
+        name: deploy-image
+      params:
+        - name: imageRef
+          value: "$(tasks.build-image.results.builtImage)"
+```
+
+
+## Design Details
+
+### finally with missing task result
+
+`Pipeline` author can specify `default` for a missing result if that `param` is optional.
+
+```yaml
+spec:
+  params:
+    - name: owner-name
+      default: "the-best-owner"
+  tasks:
+   - name: boskos-acquire
+     taskRef:
+       name: boskos-acquire
+     params:
+       - name: owner-name
+         value: $(params.owner-name)
+     results:
+      - name: leased-resource
+        value: "fake-awesome-project"
```

Comment:
i think using `value` here doesn't quite imply the defaulting behavior we're adding - maybe `default` ?